advanced-usage.md

```diff
@@ -325,6 +325,20 @@ for (std::future<cpr::Response>& fr: container) {
 ```
 {% endraw %}
 
+Asynchronous requests can also be performed using a `cpr::Session` object. It is important to note that the asynchronous request is performed directly on the session object, modifying it in the process.
+To ensure that the lifetime of the session is properly extended, the session object used must be managed by a `std::shared_ptr`. Here is an example for an asynchronous get request which uses a session object:
```

Comment:
I don't get why this has to be a `std::shared_ptr`? This could also be a `std::unique_ptr` or I just ensure the session object is still valid until the request finishes (e.g. as a class attribute).

Reply:
ACK. Then we can keep this. Please just add the reasoning behind it to the docs, like you have explained it to me here.

Reply:
Yes, but this is then something we have to clearly state.

Reply:
I have added a commit that explains the restriction.